modules/lib/src/main/resources/assets/js/app/AppWrapper.ts

```diff
@@ -310,6 +327,36 @@ export class AppWrapper
             handler(item);
         });
     }
+
+    private hideWidget(key: string, widgetData: MenuWidgetInjectionResult): void {
+        widgetData.widgetContainer.hide();
+
+        if (!this.isInternalWidget(key) && widgetData.isEnabled) {
+            widgetData.linkElements.forEach((linkElement: HTMLLinkElement) => {
+                linkElement.disabled = true;
+            });
+
+            widgetData.isEnabled = false;
+
+            // doing nothing with script elements as they can't be disabled
+        }
+    }
+
+    private showWidget(key: string, widgetData: MenuWidgetInjectionResult): void {
+        widgetData.widgetContainer.show();
+
+        if (!this.isInternalWidget(key) && !widgetData.isEnabled) {
+            widgetData.linkElements.forEach((linkElement: HTMLLinkElement) => {
+                linkElement.disabled = false;
+            });
+
+            widgetData.isEnabled = true;
+        }
+    }
```

Comment:
This can be replaced with one method accepting a boolean parameter for active/inactive. It should also be renamed since it doesn't only shows/hides but also enables/disables elements.

```Typescript
private setWidgetActive(key: string, widgetData: MenuWidgetInjectionResult, active: boolean): void {
    widgetData.widgetContainer.setVisible(active);

    if (!this.isInternalWidget(key) && widgetData.isEnabled !== active) {
        widgetData.linkElements.forEach((linkElement: HTMLLinkElement) => {
            linkElement.disabled = !active;
        });

        widgetData.isEnabled = active;
    }
}
```

Reply:
Updated